### ISSUE TITLE
Go version missing from documentations.

### ISSUE THREAD

**Issue by MonishAkram** [hashicorp/go-multierror#37]
I'm using golang v1.12 but this package is not supported because of errors.Is and errors.As functions which are not present in v1.12 and was added in v1.13. And there is no version requirement mentioned in the documentation. for this package

**Comment by IngCr3at1on** [comment]
Counterpoint: the go team themselves doesn't even support farther back than 2 versions, 1.14 was released in February which means even at the time of writing this issue 1.12 was not officially supported by the go team; is there a reason this package should support more than what is officially supported (assuming no, there is no reason to assume anything before 1.13 would work).

**Comment by webmaster777** [comment]
Having random errors like

```
> go get -u github.com/mattermost/mmctl
# github.com/mattermost/mmctl/vendor/github.com/hashicorp/go-multierror
go/src/github.com/mattermost/mmctl/vendor/github.com/hashicorp/go-multierror/multierror.go:112:9: undefined: errors.As
go/src/github.com/mattermost/mmctl/vendor/github.com/hashicorp/go-multierror/multierror.go:117:9: undefined: errors.Is
```

How does one know you need to upgrade go? I have no idea how package mangagement works in Go, but I'm used to version constraint explanations in other langs/package managers (js/php) etc.

**Comment by mwhooker** [comment]
Hmm, yeah this should be documented. FWIW, v1.0.0 supports go < 1.13, while  v1.1.0 requires go 1.13.

I also notice that in the go.mod file, the line `go 1.14`. This indicates that the module requires go 1.14. Infact, when I build this using go 1.12, I get the following error:

```
root@f97226dc12fd:/go/src/github.com/mwhooker/testmultierror# go run main.go
go: finding github.com/hashicorp/go-multierror v1.1.0
go: downloading github.com/hashicorp/go-multierror v1.1.0
go: extracting github.com/hashicorp/go-multierror v1.1.0
go: finding github.com/hashicorp/errwrap v1.0.0
go: downloading github.com/hashicorp/errwrap v1.0.0
go: extracting github.com/hashicorp/errwrap v1.0.0
# github.com/hashicorp/go-multierror
/go/pkg/mod/github.com/hashicorp/go-multierror@v1.1.0/multierror.go:112:9: undefined: errors.As
/go/pkg/mod/github.com/hashicorp/go-multierror@v1.1.0/multierror.go:117:9: undefined: errors.Is
**note: module requires Go 1.14**
```

(emphasis mine). 

Of course, because there's this ticket, this doesn't seem sufficient, but the  requirement does exist. I would be happy to review any PRs that came in for documenting the  required go version

**Comment by ggirtsou** [comment]
Fixed after upgrading to Go 1.15. I was importing `spf13/viper` and `golang-migrate` packages in my Go vendor directory  (they require different version of `go-multierror`).
